src/Microsoft.Data/StringColumn.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.Data
+{
+    public partial class StringColumn : BaseColumn
+    {
+        internal IList<IList<string>> stringBuffers = new List<IList<string>>(); // To store more than intMax number of strings
+
+        public StringColumn(string name, long length) : base(name, length, typeof(string))
+        {
+        }
+
+        public StringColumn(string name, IEnumerable<string> values) : base(name, 0, typeof(string)) {
+            values = values ?? throw new ArgumentNullException(nameof(values));
+            if (stringBuffers.Count == 0)
+            {
+                stringBuffers.Add(new List<string>());
+            }
+            var curList = stringBuffers[stringBuffers.Count - 1];
+            foreach (var value in values)
+            {
+                if (curList.Count == int.MaxValue)
+                {
+                    curList = new List<string>();
+                    stringBuffers.Add(curList);
+                }
+                curList.Add(value);
+                Length++;
+            }
+        }
+
+        private int GetBufferIndexContainingRowIndex(ref long rowIndex)
+        {
+            if (rowIndex > Length)
+            {
+                throw new ArgumentOutOfRangeException($"Index {rowIndex} cannot be greater than the Column's Length {Length}");
+            }
+            int curArrayIndex = 0;
+            int numBuffers = stringBuffers.Count;
+            while (curArrayIndex < numBuffers && rowIndex > stringBuffers[curArrayIndex].Count)
+            {
+                rowIndex -= stringBuffers[curArrayIndex].Count;
+                curArrayIndex++;
+            }
+            return curArrayIndex;
+        }
+
+        public override object this[long rowIndex]
+        {
+            get
+            {
+                int bufferIndex = GetBufferIndexContainingRowIndex(ref rowIndex);
+                return stringBuffers[bufferIndex][(int)rowIndex];
+            }
+            set
+            {
+                if (value is string)
+                {
+                    int bufferIndex = GetBufferIndexContainingRowIndex(ref rowIndex);
+                    stringBuffers[bufferIndex][(int)rowIndex] = (string)value;
+                }
+                else
+                {
+                    throw new ArgumentException(nameof(value));
```

Comment:
This should be a better exception - it should tell the user what about the argument caused the exception - basically that it wasn't a string.